packages/dds/tree/src/shared-tree/treeCheckout.ts

```diff
@@ -728,6 +732,57 @@ export class TreeCheckout implements ITreeCheckoutFork {
 		return { age: revertAge };
 	}
 
+	/**
+	 * Recursively creates a revertible for the given commit and its ancestors.
+	 */
+	private forkRevertible(
+		view: TreeCheckout,
+		commit: GraphCommit<SharedTreeChange>,
+		data: CommitMetadata,
+	): Revertible {
+		const forkRevertibleFunction = this.forkRevertible.bind(this);
+		// eslint-disable-next-line @typescript-eslint/no-this-alias
+		const outerThis = this;
+
+		const forked: Revertible = {
+			get status(): RevertibleStatus {
+				const revertibleCommit = outerThis.revertibleCommitBranches.get(commit.revision);
+				return revertibleCommit === undefined
+					? RevertibleStatus.Disposed
+					: RevertibleStatus.Valid;
+			},
+			revert: (release: boolean = true) => {
+				if (forked.status === RevertibleStatus.Disposed) {
+					throw new UsageError("Unable to revert a revertible that has been disposed.");
+				}
+
+				const revertMetrics = view.revertRevertible(commit.revision, data.kind);
+				view.logger?.sendTelemetryEvent({
+					eventName: TreeCheckout.revertTelemetryEventName,
+					...revertMetrics,
+				});
+
+				if (release) {
+					forked.dispose();
+				}
+			},
+			fork(newView: TreeCheckout): Revertible {
+				return forkRevertibleFunction(newView.fork(), commit, data);
```

Comment:
Are commit and data immutable? If these objects are shared between forks, is there any risk of modification that could affect the original state?